src/v/redpanda/debug_bundle.cc

```diff
@@ -11,16 +11,123 @@
 
 #include "redpanda/debug_bundle.h"
 
+#include "ssx/future-util.h"
+#include "utils/gate_guard.h"
 #include "vlog.h"
 
+#include <seastar/http/exception.hh>
 #include <seastar/util/log.hh>
+#include <seastar/util/process.hh>
+
+#include <fmt/format.h>
 
 static ss::logger bundle_log{"debug_bundle"};
 
-debug_bundle::debug_bundle(const std::filesystem::path& write_dir)
-  : _write_dir{write_dir} {}
+using consumption_result_type
+  = debug_bundle::rpk_consumer::consumption_result_type;
+using stop_consuming_type = debug_bundle::rpk_consumer::stop_consuming_type;
+using tmp_buf = stop_consuming_type::tmp_buf;
+
+ss::future<consumption_result_type>
+debug_bundle::rpk_consumer::operator()(tmp_buf buf) {
+    std::string str_buf{buf.begin(), buf.end()}, line;
+    vlog(bundle_log.trace, "Consumer read {}", str_buf);
+    std::stringstream ss(str_buf);
+    while (!ss.eof()) {
+        std::getline(ss, line);
+        if (line.starts_with("Debug bundle saved to")) {
+            vlog(bundle_log.trace, "Stop condition reached on line {}", line);
+            return make_ready_future<consumption_result_type>(
+              stop_consuming_type({}));
+        }
+    }
+    return make_ready_future<consumption_result_type>(ss::continue_consuming{});
+}
+
+debug_bundle::debug_bundle(
+  const std::filesystem::path& write_dir, const std::filesystem::path& rpk_path)
+  : _write_dir{write_dir}
+  , _in_progress_filename{"debug-bundle.zip"}
+  , _rpk_cmd{rpk_path} {}
 
 ss::future<> debug_bundle::stop() {
     vlog(bundle_log.info, "Stopping debug bundle ...");
+    co_await _rpk_gate.close();
+}
+
+ss::future<> debug_bundle::start_creating_bundle() {
+    if (ss::this_shard_id() != debug_bundle_shard_id) {
+        return container().invoke_on(
```

Comment:
container().invoke_on has a fast path for already being on the same core. so you can run the invoke_on in all cases. to do that, split out the body of the function to avoid the recursion that would result.